Makefile

```diff
@@ -8,53 +10,66 @@ help:
 build: ## Build project
 	dotnet build -f=netcoreapp2.0
 
-.PHONY: test
-test: ## Run unit, authentication, integration tests
-	make unit-test
-	make authentication-test
-	make integration-test
-
-.PHONY: authentication-test
-authentication-test: ## Run authentication tests
-	dotnet test ./src/Notify.Tests/Notify.Tests.csproj --no-build -f=netcoreapp2.0 --filter TestCategory=Unit/AuthenticationTests
-
 .PHONY: integration-test
 integration-test: ## Run integration tests
 	dotnet test ./src/Notify.Tests/Notify.Tests.csproj --no-build -f=netcoreapp2.0 --filter TestCategory=Integration
 
 .PHONY: unit-test
 unit-test: ## Run unit tests
-	dotnet test ./src/Notify.Tests/Notify.Tests.csproj --no-build -f=netcoreapp2.0 --filter TestCategory=Unit/NotificationClient
+	dotnet test ./src/Notify.Tests/Notify.Tests.csproj --no-build -f=netcoreapp2.0 --filter TestCategory=Unit
 
 .PHONY: single-test
 single-test: ## Run a single test: make single-test test=[test name]
 	dotnet test ./src/Notify.Tests/Notify.Tests.csproj --no-build -f=netcoreapp2.0 --filter $(test)
 
 .PHONY: build-single-test
-build-single-test: ## Run a single test: make single-test test=[test name]
+build-single-test: ## Build and run a single test: make single-test test=[test name]
 	dotnet test ./src/Notify.Tests/Notify.Tests.csproj -f=netcoreapp2.0 --filter $(test)
 
 .PHONY: build-test
-build-test: ## Build and test
-	make build
-	make unit-test
-	make authentication-test
-	make integration-test
-
-.PHONY: build-integration-test
-build-integration-test: ## Build and integration test
-	make build
-	make integration-test
+build-test: build unit-test integration-test ## Build, test, integration test
 
 .PHONY: build-unit-test
-build-unit-test: ## Build and integration test
-	make build
-	make unit-test
+build-unit-test: build unit-test ## Build and unit test
 
 .PHONY: build-release
 build-release: ## Build release version
 	dotnet build -c=Release -f=netcoreapp2.0
 
+.PHONY: generate-env-file
+generate-env-file: ## Generate the environment file for running the tests inside a Docker container
+	scripts/generate_docker_env.sh
+
+.PHONY: test-with-docker
+test-with-docker: generate-env-file ## Test with docker
+	docker build -t ${DOCKER_BUILDER_IMAGE_NAME} .
+	docker run -it --rm \
+		--name "${USER}-notifications-net-client-manual-test" \
+		-v "`pwd`:/var/project" \
+		--env-file docker.env \
+		${DOCKER_BUILDER_IMAGE_NAME} \
+		make build-unit-test
```

Comment:
We've got two slightly conflicting principles in this make file

In this one, make `test-with-docker` will just run the unit tests but further up we are explicit like `make unit-test`. We also have `build-test` which will run both unit and integration.

Personally, I prefer `make test` to just run unit tests and `make integration-tests` to just run integration tests and this fits with what concourse is doing I feel. I don't feel too strongly which but I feel a bit more consistency could be good.

Reply:
sensible, changed to reflect yr naming. also removed some more stuff